woocommerce/woocommerce-android
# [Kotlin Warnings As Errors] All Modules - Enable All Warnings as Errors

Parent: #6816
Closes: #6820



### Description


This PR enabled all warnings as error for all modules.

This `allWarningsAsErrors` configuration was previously applied on the module level in order to make sure that, as the overall `Kotlin Warnings as Errors` work is progressing, no new warnings are added to a specific module, which is already free of warnings.

However, now that the overall `Kotlin Warnings as Errors` work is complete on all modules, this module level `allWarningsAsErrors` configuration is now replaced by a root level `allWarningsAsErrors` configuration that will be applied by default to all modules.

Also, as part of this change, the whole `kotlinOptions` configuration is now moved to the root level so that it is being applied by default to all modules from now on.

-----

PS: @malinajirka I added you as the main reviewer, that is, in addition to the @woocommerce/owl-team team itself, but randomly, since I just wanted someone from the `Woo Android` team to sing-off on that change for WCAndroid as well.

### Testing instructions


- There is nothing much to test here.
- Verifying that all the CI checks are successful should be enough.
- If you really want to be thorough, try to add an artificial warning per source (`main`, `test`, `androidTest`), per module (`WooCommerce`, `cardreader`, `quicklogin`) and see the `assemble` related task fail. Then, remove this warning and see it succeeding again. 

### Merge instructions

- [x] Wait for #6902 to be approved and merged to `trunk`.
- [x] Remove `status: do not merge` label.

### Images/gif
